### PR TITLE
enable tab compatbility

### PR DESCRIPTION
**Pull request by adrianjost**
This is a basic implementation to allow tab usage. I handle each tab like a single space character. In most cases this should be fine as it's really rare that tabs and spaces are mixed.

Issue: https://github.com/dmnd/dedent/issues/10

### CODE REVIEW

**Comment by adrianjost** [comment]
@dmnd could you give an estimation, when this pull might get merged?

**Comment by adrianjost** [comment]
If someone is finding this pr and is looking for a fast fix: I published a version with my fix on [npm](https://www.npmjs.com/package/dedent-tabs).

File: dedent.js [hunk 1]
```diff
@@ -39,7 +39,8 @@ export default function dedent(
 
   if (mindent !== null) {
     const m = mindent; // appease Flow
-    result = lines.map(l => l[0] === " " ? l.slice(m) : l).join("\n");
+    const whitespaceCharacters = [" ", "\t"]; // [space, tab]
+    result = lines.map(l => whitespaceCharacters.includes(l[0]) ? l.slice(m) : l).join("\n");
```
Comment (aleclarson) [inline dedent.js]: Doing `l[0] === " " || l[0] === "\t"` is probably faster

**Review by JakeChampion — APPROVED** [review]

**Comment by sorrykim** [comment]
when is it going to be merged?

**Comment by adrianjost** [comment]
@sorrykim I wouldn't expect this to be merged in a timely manner. If you really need this feature, I would recommend you using [dedent-tabs](https://www.npmjs.com/package/dedent-tabs) which is a fork of this project I still maintain and has this feature built in.

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 this looks great, thanks @adrianjost!

**Comment by fisker** [comment]
That's fast.

**Pull request merged**